OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -2243,6 +2243,17 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				// ChargeTime is now replaced by ChargeDelay.
+				// ChargeDelay uses 500 as a default value from ChargeTime 25.
+				if (engineVersion < 20151021)
+				{
+					if (depth == 2 && parentKey == "PortableChrono" && node.Key == "ChargeTime")
+						node.Key = "ChargeDelay";
+
+					if (depth == 2 && parentKey == "PortableChrono" && node.Key == "ChargeTime" && node.Value.Value != null)
+						node.Value.Value = (Convert.ToInt32(node.Value.Value) * 25).ToString();
```

Comment:
Use `Exts.ParseIntegerInvariant(node.Value.Value)` instead of `Convert.ToInt32(node.Value.Value)`.